apps/arweave/src/ar_tx.erl

```diff
@@ -431,6 +432,13 @@ chunks_to_size_tagged_chunks(Chunks) ->
 sized_chunks_to_sized_chunk_ids(SizedChunks) ->
 	[{ar_tx:generate_chunk_id(Chunk), Size} || {Chunk, Size} <- SizedChunks].
 
+tx_format_to_http_header(TXFormat) ->
+	[{<<"x-tx-format">>, tx_http_format(TXFormat)}].
+
+tx_http_format(?WITH_TX_DATA) -> ?TX_WITH_DATA_HTTP_FORMAT;
```

Comment:
Looks like, this function can be rewritten as:
```erlang
tx_http_format(?WITH_TX_HEADER) -> ?TX_WITHOUT_DATA_HTTP_FORMAT;
tx_http_format(_) -> ?TX_WITH_DATA_HTTP_FORMAT.
```

Reply:
Indeed. But we're only making this explicit so that the intent is clear that we're handling what we know and what we expect in each clause. This is intentional. A handle all clause definitely works, but will also unnecessarily hide & abstract the tx-format protocol which we're expecting.